translate/storage/test_properties.py

```diff
@@ -1104,7 +1109,13 @@ def test_cleaning_attributes(self):
         generatedcontent = BytesIO()
         propfile.settargetlanguage("fr")
         propfile.serialize(generatedcontent)
-        expected_xml = properties.XWikiPageProperties.XML_HEADER + """<xwikidoc reference="XWiki.AdminTranslations">
+
+        ## ElementTree changed the way to process the attribute between Python 3.7 and 3.8
+        if sys.version_info.minor >= 8:
+            xml_open = "<xwikidoc reference=\"XWiki.AdminTranslations\" locale=\"fr\">"
+        else:
+            xml_open = "<xwikidoc locale=\"fr\" reference=\"XWiki.AdminTranslations\">"
```

Comment:
Maybe it's better to compare XML instead of guessing ordering of the attributes? Maybe something like https://stackoverflow.com/a/7060342/225718 would be useful here.

Reply:
I hesitated about that, could be useful if I had more tests with various XML attributes but here there's only those two (I managed to remove the other ones from the test to avoid the problem). I'm in favor of keeping it like that since it's working properly on different platforms / versions of Python.

Reply:
Looking again at this I might realized why other formats do not need this - everything else uses lxml. One of the reasons is that it does better job in preventing some security issues, see https://docs.python.org/3/library/xml.html#xml-vulnerabilities.

Can you please change the XWiki parser to use it as well? It should also disable entities resolving to avoid vulnerability in that. This can be done by using `etree.XMLParser(resolve_entities=False)`. My guess is that this condition will no longer be necessary with lxml.

Reply:
> Can you please change the XWiki parser to use it as well?

ok I will take a look to use lxml. Hope the switch will be easy :)